### ISSUE TITLE
Toast Stack + Toast Item

### ISSUE THREAD

**Issue by claviska** [shoelace-style/webawesome#105]
As part of #104, alerts will no longer be the thing that enables toast notifications. Toasts were shimmed into alerts and need accessibility improvements anyways — so let's do it right this time.

Moving forward, we will add `<wa-toast>` and `<wa-toast-item>` to the library and provide utility functions to initiate toast notifications.

### Open questions

- Do we have a rough design for what toasts will look like?

/cc @lindsaym-fa

**Comment by steunix** [comment]
Just out of curiosity, will this component be included in one of the next betas or will they be added later in the 3.x stable cycle? It's the last bit of WA I'm waiting and longing for ;)

**Comment by claviska** [comment]
I can see them shipping in a beta version. I don't have a definite timeline for this one just yet, though!

**Comment by steunix** [comment]
> I can see them shipping in a beta version. I don't have a definite timeline for this one just yet, though!

That's great. Keep up the great work!

**Comment by lindsaym-fa** [comment]
re: design, I'm imagining slots for `icon`, content (default), and `action` with a close button + progress ring to show a countdown of when the toast will close automatically, along with a `without-close` attribute to optionally hide the close button.

Something like this:

<img width="420" height="117" alt="default toast with a subtle outlined style, box shadow, and minimal brand coloring" src="https://github.com/user-attachments/assets/5a7be5ac-c601-4af5-a08b-1e9cd0dde1c3" />

With `variant=""` and `appearance=""` options:

<img width="808" height="653" alt="toasts in brand, neutral, success, warning, and danger variants followed by outlined, filled, filled + outlined, and accent appearances" src="https://github.com/user-attachments/assets/bd386184-28db-44dd-b9fb-d22beaf0f373" />

Here's a rough pen for the styles: https://codepen.io/omnifides/pen/XJmRQmp

h/t to @claviska for the inspiration of surrounding the close button with the progress indicator. 🌟

**Comment by ricky11** [comment]
Toast is what we're waiting for before we can think of deploying this to prod, what can we use in replacement of it, wonder why it was left out as core component in most ui's.

**Comment by GedMarc** [comment]
I added one into the angular-awesome npm package in the meantime if you are using angular
Also for us, its considered a base component hehe

**Comment by ricky11** [comment]
using vue 😁, and need my toast.

**Comment by aaronpinero** [comment]
I was just reading GitHub’s note about not using toasts and suggesting alternatives that are more accessible. Will the wa-toast address some of the accessibility concerns?

**Comment by steunix** [comment]
Will the toast be a Pro feature, then?

**Comment by claviska** [comment]
Yes, the team has decided that toast will be released under pro.

**Comment by kasper-i** [comment]
What was the reason for putting the toast component under Pro? I'm wondering since I'm transitioning from Shoelace which included it.

**Comment by claviska** [comment]
The "toast" in Shoelace was a hack added to `<sl-alert>` with poor accessibility that also convoluted the alert itself (e.g. requiring an `open` attribute to show, which has been a huge foot gun).

The toast we're adding to Web Awesome is substantially better than what you saw in Shoelace. It involves two separate components (toast + toast item) and improves the overall API and accessibility in comparison.

We're releasing it as Pro partly because of the level of work it requires. Consider that there are entire libraries devoted to this functionality. More complexity generally means it goes into Pro.

Considering we've released over 50 free components with more on the way, Pro is a reasonable place for toast + toast item to exist to help sustain the project and the team that's working on it.

I realize this is an unpopular opinion to free users. But if you're finding value in Web Awesome, I respectfully ask you to consider buying Pro so we can keep building and adding to the project.

Thanks for supporting us!

**Comment by CiTroNaK** [comment]
I am looking forward to it. 

I like the `action` slot a lot. It fits well with what I built years ago using [Tailwind in a Rails app](https://petr.codes/blog/rails/modern-rails-flash-messages/part-1/). 

<details><summary>Examples</summary>
<img width="412" height="399" alt="Image" src="https://github.com/user-attachments/assets/f2a8a0f9-0ae8-4ad4-a994-cf0b1bf5e45a" />

[gif version](https://dev-to-uploads.s3.amazonaws.com/i/3p2sp78pute2vimhj5ui.gif)
</details>

**Comment by lbird-mma** [comment]
Also waiting for this one to complete, before I will migrate from Shoelace to WebAwesome. I'm a heavy user of toasts with sl-alert
